src/auth.ts

```diff
@@ -548,29 +553,73 @@ export class BasicAuth extends AuthType implements IBasicAuthType {
   }
 }
 
+/**
+ * hidden
+ */
 export abstract class NativeAuthType {
   protected config: IConfig;
   protected client: IClient;
+  protected userService: ISingleUserService;
+  private tokenContext: ICombinedTokenContext;
+  private emitter: IEventEmitter;
+  private authToken: string;
 
-  constructor(deps: AuthTypeDependencies) {
+  constructor(deps: NativeAuthDependencies) {
     this.config = deps.config;
     this.client = deps.client;
+    this.userService = deps.userService;
+    this.tokenContext = deps.tokenContext;
+    this.emitter = deps.emitter;
   }
+
+  /**
+   * Get the raw auth token of the active user from local storage.
+   */
+  public getToken(): string {
+    return this.tokenContext.get();
+  }
+
+  /**
+   * @hidden
+   */
+  public storeToken(token: string) {
+    let originalToken = this.authToken;
+    this.authToken = token;
+    this.tokenContext.store(this.authToken, {'permanent': true});
+    this.emitter.emit('auth:token-changed', {'old': originalToken, 'new': this.authToken});
+  }
+
 }
 
-export class GoogleNativeAuth extends NativeAuthType {
-  public authenticate(data?: any, options?: any): Promise<any> {
+/**
+ * GoogleNativeAuth handles logging into googleplus through the cordova-plugin-googleplus plugin.'
+ * @featured
+ */
+export class GoogleAuth extends NativeAuthType implements IGoogleAuth {
+  public login(scope: GoogleScopes[]): Promise<any> {
```

Comment:
Does it make sense for these scopes to be a parameter here, or should it be part of the auth config?